libvmi/CMakeLists.txt

```diff
@@ -146,9 +146,10 @@ if (ENABLE_KVM)
                 target_include_directories(vmi_shared PRIVATE ${Libkvmi_INCLUDE_DIRS})
             endif ()
         endif ()
-        target_include_directories(vmi_shared PRIVATE ${JSON-C_INCLUDE_DIRS})
+        target_include_directories(vmi_shared PUBLIC ${JSON-C_INCLUDE_DIRS})
         # CMAKE_DL_LIBS -> dlopen* lib
-        target_link_libraries(vmi_shared PRIVATE ${CMAKE_DL_LIBS} ${JSON-C_LIBRARIES})
+        target_link_libraries(vmi_shared PRIVATE ${CMAKE_DL_LIBS})
+        target_link_libraries(vmi_shared PUBLIC ${JSON-C_LIBRARIES})
         list(APPEND VMI_PUBLIC_HEADERS events.h)
         list(APPEND VMI_PUBLIC_DEPS ${CMAKE_DL_LIBS} ${JSON-C_LIBRARIES})
```

Comment:
Any idea why `libdl.so` is listed as a public dependency? As far as I am aware it should be private, right?

Reply:
Indeed, it shouldn't be a public dependency, we only use dlopen internally
https://github.com/libvmi/libvmi/blame/master/libvmi/CMakeLists.txt#L150